api/src/main/java/org/ehcache/spi/sizeof/SizeOfEngine.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.spi.sizeof;
+
+/**
+ * SizeOf engines are used to calculate the size of objects.
+ * 
+ * @author Abhilash
+ *
+ */
+public interface SizeOfEngine {
+  
+  /**
+   * Size of the objects on Heap
+   * 
+   * @param objects objects to be sized
+   * @return size of the objects on heap 
+   */
+  long sizeof(Object... objects);
+  
+  /**
+   * Sizes the key along with the offset of 
+   * the key holder
+   * 
+   * @param key key to be sized
+   * @return size of the key along with the offset
+   */
+  long sizeofKey(Object key);
+  
+  /**
+   * Sizes the value along with the offset of 
+   * the value holder 
+   * 
+   * @param value value to be sized
+   * @return size of the value along with the offset
+   */
+  long sizeofValue(Object value);
+  
+  /**
+   * Bytes required on adding each key-value pair
+   * 
+   * @return size overhead of adding a key-value pair
+   */
+  long getCHMOffset();
```

Comment:
There must be a better way to abstract that mechanism. We should meet around this.